src/OrchardCore.Modules/OrchardCore.Media.Azure/Services/ImageSharpImageCacheOptionsConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.Media.Azure.Helpers;
+
+namespace OrchardCore.Media.Azure.Services;
+
+internal class ImageSharpImageCacheOptionsConfiguration : IConfigureOptions<ImageSharpImageCacheOptions>
+{
+    private readonly IShellConfiguration _shellConfiguration;
+    private readonly ShellSettings _shellSettings;
+    private readonly ILogger _logger;
+    private readonly FluidParserHelper<ImageSharpImageCacheOptions> _fluidParserHelper;
+
+    public ImageSharpImageCacheOptionsConfiguration(
+        IShellConfiguration shellConfiguration,
+        ShellSettings shellSettings,
+        ILogger<ImageSharpImageCacheOptionsConfiguration> logger)
+    {
+        _shellConfiguration = shellConfiguration;
+        _shellSettings = shellSettings;
+        _logger = logger;
+        _fluidParserHelper = new(shellSettings);
+    }
+
+    public void Configure(ImageSharpImageCacheOptions options)
+    {
+        var section = _shellConfiguration.GetSection("OrchardCore_Media_Azure_ImageSharp_Cache");
+
+        options.ContainerName = section.GetValue(nameof(options.ContainerName), string.Empty);
+        options.ConnectionString = section.GetValue(nameof(options.ConnectionString), string.Empty);
+        options.CreateContainer = section.GetValue(nameof(options.CreateContainer), true);
+        options.RemoveContainer = section.GetValue(nameof(options.RemoveContainer), false);
+
+        try
+        {
+            // Container name must be lowercase.
+            options.ContainerName = _fluidParserHelper.ParseAndFormat(options.ContainerName).ToLower();
+        }
+        catch (Exception ex)
+        {
+            _logger.LogCritical(ex, "Unable to parse Azure Media ImageSharp Image Cache container name.");
+            throw;
```

Comment:
throwing exception here will cause the startup of the app to fail. I would suggest not to throw exception during configuration.

Reply:
Then we should not throw and break the startup.

Reply:
I mean, I agree with the current implementation of throwing. The tenant can't possibly do anything useful if Media is broken, and startup should break.

Reply:
So we crash the app? One time I we had a discussion about caching images. I always thought the app should be able to run without caching "It would be slow" but should run. I don't recall the exact details.

Anyway, if this is what we do today, the it is what it is.

Reply:
Yep, and I'd argue crashing the app is the right thing to do. Media access not being configured is only slightly better than DB access not being configured, but both basically completely break the app.

Note that I'm still talking about _configuration_ being wrong. The Media storage (or the DB) having issues and being transiently unavailable is of course something we should (and do) handle more gracefully. But configuration issues should be addressed immediately.

We discussed caching under https://github.com/OrchardCMS/OrchardCore/issues/14859. With the current implementation, copying remote Media files (like from Blob Storage) to the local webroot is required to serve them; without this step, Media files won't be accessible, not just slower to access. Before this PR, the same was necessary for the ImageSharp cache files too (that's a different request pipeline though, just the same concept).